lib/puppet/provider/archive/ruby.rb

```diff
@@ -31,7 +31,11 @@ def exists?
   end
 
   def create
-    transfer_download(archive_filepath) unless checksum?
+    if resource[:ensure] == 'latest' and resource[:source] =~ /^s3/
+      transfer_download(archive_filepath) unless s3_checksum?
+    else
+      transfer_download(archive_filepath) unless checksum?
+    end
```

Comment:
Don't change it here, either modify the default checksum? method or implement your own s3 provider with your custom checksum? method.